iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapred/MapredIcebergInputFormat.java

```diff
@@ -75,8 +75,13 @@ public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
   @Override
   public RecordReader<Void, Container<T>> getRecordReader(InputSplit split, JobConf job,
                                                           Reporter reporter) throws IOException {
-    IcebergSplit icebergSplit = ((IcebergSplitContainer) split).icebergSplit();
-    return new MapredIcebergRecordReader<>(innerInputFormat, icebergSplit, job, reporter);
+    try {
+      org.apache.hadoop.mapreduce.InputSplit inputSplit = split instanceof IcebergMergeSplit ?
```

Comment:
what's the purpose of this ternary operator if you can simply cast to `InputSplit`?

Reply:
Normal iceberg splits are retrieved using ((IcebergSplitContainer) split).getIcebergSplit(). Hence we need to check whether the split is IcebergMergeSplit ot IcebergSplit.

Reply:
why do you need `(IcebergMergeSplit) split` cast if it's already `InputSplit` type?

Reply:
Because this is still not of `InputSplit` type. Casting it to IcebergMergeSplit will allow this to be easily casted.